.travis/script.sh

```diff
@@ -1,40 +1,35 @@
-#!/usr/bin/env sh
-set -v
+#!/usr/bin/env bash
+# coding=utf-8
+set -veuo pipefail
 
-flake8 --config flake8.cfg || exit 1
-
-result=0
+flake8 --config flake8.cfg
 
 export DJANGO_SETTINGS_MODULE=pulpcore.app.settings
 
+# Perform migrations.
 pulp-manager migrate auth --noinput
 pulp-manager makemigrations pulp_app --noinput
 pulp-manager makemigrations pulp_ansible
 pulp-manager migrate --noinput
-if [ $? -ne 0 ]; then
-  result=1
-fi
 
-pushd ../pulp
-coverage run manage.py test pulp_ansible.tests.unit
-if [ $? -ne 0 ]; then
-  result=1
-fi
-popd
+# Run unit tests.
+(cd ../pulp && coverage run manage.py test pulp_ansible.tests.unit)
 
+# Run functional tests.
 pulp-manager reset-admin-password --password admin
 pulp-manager runserver >>~/django_runserver.log 2>&1 &
 rq worker -n 'resource_manager@%h' -w 'pulpcore.tasking.worker.PulpWorker' >> ~/resource_manager.log 2>&1 &
 rq worker -n 'reserved_resource_worker_1@%h' -w 'pulpcore.tasking.worker.PulpWorker' >> ~/reserved_worker-1.log 2>&1 &
-
 sleep 5
-pytest -v --color=yes --pyargs pulp_ansible.tests.functional
-
-if [ $? -ne 0 ]; then
-  result=1
-  cat ~/django_runserver.log
-  cat ~/resource_manager.log
-  cat ~/reserved_worker-1.log
-fi
-
-exit $result
+show_logs_and_return_non_zero() {
+    readonly local rc="$?"
```

Comment:
If pytest returns a non-zero exit code, then we capture it here, and re-raise the same exit code later. This is much nicer than `return 1`, which masks pytest's exit code.